catalog/rocks/entry_catalog.go

```diff
@@ -202,6 +358,19 @@ func (e *EntryCatalog) Diff(ctx context.Context, repositoryID graveler.Repositor
 }
 
 func (e *EntryCatalog) GetEntry(ctx context.Context, repositoryID graveler.RepositoryID, ref graveler.Ref, path Path) (*Entry, error) {
+	if err := Validate("repositoryID", repositoryID, ValidateRepositoryID); err != nil {
+		return nil, err
+	}
+	if err := Validate("ref", ref, ValidateRef); err != nil {
+		return nil, err
+	}
+	if err := Validate("path", path, ValidatePath); err != nil {
+		return nil, err
+	}
+	if err := Validate("path", path.String(), ValidateNonEmptyString); err != nil {
```

Comment:
Here and elsewhere: 
Calling `ValidateNonEmptyString` instead of `ValidatePath`.